include/z64.h

```diff
@@ -346,19 +346,19 @@ typedef struct {
 
 typedef struct {
     /* 0x00 */ char  unk_00[0x4];
-    /* 0x04 */ void* segment;
-    /* 0x08 */ u8    state;
-    /* 0x0C */ f32   unk_0C;
-    /* 0x10 */ u16   frames;
-    /* 0x12 */ u16   unk_12;
-    /* 0x14 */ s32   subCamId;
-    /* 0x18 */ u16   unk_18;
-    /* 0x1A */ u8    unk_1A;
-    /* 0x1B */ u8    unk_1B;
-    /* 0x1C */ CutsceneCameraPoint* subCamLookAtPoints;
-    /* 0x20 */ CutsceneCameraPoint* subCamEyePoints;
-    /* 0x24 */ CsCmdActorAction* linkAction;
-    /* 0x28 */ CsCmdActorAction* npcActions[10]; // "npcdemopnt"
+    /* 0x04 */ void* scriptPtr;
+    /* 0x08 */ u8 state;
+    /* 0x0C */ f32 timer;
+    /* 0x10 */ u16 curFrame; // current frame of the script that is running
+    /* 0x12 */ u16 unk_12; // set but never used
+    /* 0x14 */ s32 subCamId;
+    /* 0x18 */ u16 camEyePointsAppliedFrame; // when set, indicates that the cam `eye` data was applied
+    /* 0x1A */ u8 camAtReady; // cam `at` data is ready to be applied
+    /* 0x1B */ u8 camEyeReady; // cam `eye` data is ready to be applied
+    /* 0x1C */ CutsceneCameraPoint* camLookAtPoints;
+    /* 0x20 */ CutsceneCameraPoint* camEyePoints;
```

Comment:
Like a regression, all subCam fields were renamed to subCamId/subCamSomething, here subCamId is kept but fields become just camSomething

Reply:
I dont think the fact that its a subcam is relevant in this context. Its moreso describing that its a list of points for the spline, and not as concerned with what cam or subcam it belongs to

Reply:
It is specifically applied in the spline-command code to the camera stored in `csCtx->subCamId`, which is directly set to a SubCamera `csCtx->subCamId = Play_CreateSubCamera(play);`, but that can also be addressed in separate camera convos outside this PR

Reply:
if the issue at hand is "cam" being conflated with "main cam", would it help if we just removed "cam"?
the type is already CutsceneCameraPoint, so its clear its for the camera, and it could just be `atPoints` and `eyePoints`

I still feel strongly about not including "subcam" in its name, this feels entirely irrelevant to its purpose within the cutscene system

Reply:
Yeah, and I don't feel strongly about including `subCam` in its name, my only argument is consistency but if it's not needed here, then that's fine with me

Reply:
I think my preference is camXPoints over just xPoints, but if the latter helps with compromise then i am fine with it